code/common/array.c

```diff
@@ -74,13 +74,12 @@ _array_resize(array_void_t *a, size_t size, const size_t elem_size)
 
     if (a->capacity < size) {
         int ret;
-        if ((ret = _array_grow_to(a, size, elem_size)) != 0) {
+        if ((ret = _array_reserve(a, size, elem_size)) != 1)

```

Comment:
Why changing `_array_grow_to` to `_array_reserve`?

That looks wrong, but what was your case?

Reply:
So resize and reserve were broken since your recent changes; I decided to use std::string as the basic logic, so when resizing we grow the array and malloc. I think the initial problem was that `array_resize` wasn't actually resizing, or setting the size variable

Reply:
So, `array_resize` was indeed not following `std::string` (or `std::vector`) which I wanted to change recently. Likely where I introduced the bug.

But, `array_grow_to` should effectively be the same as `array_reserve`, pretty much the internal version. There is still one inaccuracy between `array_reserve` and `std::vector::reserve`. `std::vector::reserve` reserve places for N elements, instead of N additional elements. So, if the array already have elements, adding N elements should be reserved as `a.reserve(a.size() + N)`. Maybe we should change it to.

Anyhow, the `_array_grow_to` seems fine in this case, do you have the case that broke it? I'm slowly adding unit tests, because this project is severely lacking unit tests.

Reply:
I think the first issue to do with use of arrays was finding the ip address of the auth server before any authentication is made. Also had issues with the number of available bag slots and another issue around auth server handlers

Reply:
Also I think `std::vector::resize` also frees memory if the required size is less than current - didn't want to enforce that because that might cause a whole heap of more problems